src/fsharp/FSharp.Core/seq.fs

```diff
@@ -943,35 +948,35 @@ namespace Microsoft.FSharp.Collections
                 member val SeqState = SeqProcessNextStates.NotStarted with get, set
                 member __.HaltedIdx = haltedIdx
 
-                interface ISeqPipeline with
+                interface IPipeline with
                     member __.StopFurtherProcessing pipeIdx = haltedIdx <- pipeIdx
 
             // SetResult<> is used at the end of the chain of SeqComponents to assign the final value
             type SetResult<'T> (result:Result<'T>) =
-                inherit SeqConsumer<'T,'T>()
+                inherit Consumer<'T,'T>()
 
                 override __.ProcessNext (input:'T) : bool =
                     result.Current <- input
                     true
 
             type Pipeline() =
```

Comment:
Pipeline/result did used to carry more information in previous iterations. I will change the change the name.

What about the names

Terminator
OutOfBand
Halter

Any of those short and snappy (and correct!) enough?